src/loaders/ObjectLoader.js

```diff
@@ -632,11 +679,73 @@ Object.assign( ObjectLoader.prototype, {
 			if ( data.visible !== undefined ) object.visible = data.visible;
 			if ( data.userData !== undefined ) object.userData = data.userData;
 
-			if ( data.children !== undefined ) {
+			/*
+			 * SkinnedMesh creates Bone instances as its children in the constructor
+			 * so skip Bone instance creation here.
+			 */
+			if ( data.type === 'SkinnedMesh' ) {
+
+				var scope = this;
+
+				function traverse ( data ) {
+
+					if ( data.children !== undefined ) {
 
-				for ( var child in data.children ) {
+						for ( var child in data.children ) {
 
-					object.add( this.parseObject( data.children[ child ], geometries, materials ) );
+							if ( data.children[ child ].type !== 'Bone' ) {
+
+								var parent;
+
+								if( data.type === 'SkinnedMesh' ) {
+
+									parent = object;
+
+								} else if ( data.type === 'Bone' ) {
+
+									// should make a method THREE.Skeleton.findBoneByName() ?
+									for ( var i = 0, il = object.skeleton.bones.length; i < il; i ++ ) {
+
+										if ( object.skeleton.bones[ i ].name === data.name ) {
+
+											parent = object.skeleton.bones[ i ];
+											break;
+
+										}
+
+									}
+
+								}
+
+								if ( parent ) {
+
+									parent.add( scope.parseObject( data.children[ child ], geometries, materials, skeletons ) );
+
+								}
+
+							} else {
+
+								traverse( data.children[ child ] );
+
+							}
+
+						}
+
+					}
+
+				}
+
+				traverse( data );
+
+			} else {
```

Comment:
This is the very part I really wanna simplify but I couldn't...

`Bone` instances are generated for `SkinnedMesh` as its children
from `.bones` property in `SkinnedMesh` constructor.
So we have to skip `Bone` parsing here not to make duplicated `Bone`s.

But we still need to parse the children objects of `Bone`
(imagine that right hand bone has a gun mesh)
, and then make them children of `Bone` generated in the constructor.